### ISSUE TITLE
bugfix/prevent overbooking for slots

### ISSUE THREAD

**Issue by dexios1** [serenity-health/roadmap#6]
**Describe the bug**
Comment from Dr. Darko: Please alert the team to not book 2 or more people for the same time. Can the time slot be selected only once? This is causing problems at the waiting areas

**To Reproduce**
1. Start a visit for a patient and choose a time slot
2. Start a visit for another patient and choose the same time slot (that should not allowed)

**Expected behavior**
Once a patient visit is started, the slot assigned to them should not be avaialble to others unless the patient cancels the bill before their encounter is due

**Screenshots/Screencast**
N/A

**Desktop (please complete the following information):**
 - OS: Mac, Windows
 - Browser: Chrome
 - Version: Unknown

**Additional context**
N/A

**Comment by chris-dare** [comment]
Hi @seddie62, let's use the solution you proposed. If the slot status is not free then it should be disabled for selection

**Comment by seddie62** [comment]
@dexios1 https://gitlab.com/serenity-cic/emr-web/-/merge_requests/173. Merged into develop

**Comment by chris-dare** [comment]
Just tested. Looks awesome <3. Thanks @seddie62

**Comment by chris-dare** [comment]
Hey Frank and Ansbert. You'll notice now that slots that have been booked can no longer be selected as seen in the screenshot below for 12:45pm

![image](https://user-images.githubusercontent.com/18426797/148214312-afec220e-33eb-486e-aa33-e16c9b201b83.png)

**Comment by seddie62** [comment]
@dexios1 what's the way forward here

**Comment by chris-dare** [comment]
> @dexios1 what's the way forward here

Sorry, I somehow missed this.

@seddie62 I need you to run a check at the point of creating the visit to confirm that the slot has not been booked.

@TuyizereBapt, we're going to have to make some modifications to `encounter.models.Encounter.create` such that an encounter cannot be created unless the slot is free. This is already there but there needs to be a tweak. Let's discuss that at 10am tomorrow. Can you send me a calendar invite?
